core/cap-0040.md

```diff
@@ -255,14 +256,23 @@ signature for C_i, the sender who is watching the network may see D_i be
 submitted, extract the signature for C_i from the signatures on D_i, attach the
 signature to C_i, and submit C_i.
 
+The use of the payload signer in CAP-21's payment channel requires a payload of
+32 bytes to store the hash of another Stellar transaction. 32 bytes are required
+because Stellar transaction hashes utilize the SHA-256 algorithm.
+
 ### Other Uses
 
 This new signer is likely to have other applications in scenarios similar to
 where HASH_X signers are currently used, except that the data revealed would not
 only be a hash shared by multiple transactions across multiple chains, but could
-be a signature for a transaction on another chain, or a siganture for any other
-use. However, this has limited use to only signatures of ed25519 keys as
-specified.
+be a signature for a hash validated by a smart contract on another chain, or a
+siganture for any other use. However, this has limited use to only signatures of
+ed25519 keys as specified.
+
+The use of the payload signer in these other applications could require a
+variety of different payload sizes, some of which would be impractical for
+permitting without changes to Stellar's fee participation. However, a payload up
+to 64 bytes would likely be required for some hashing algorithms.
```

Comment:
Do you have some examples of when 64 bytes would be used? I'm curious if there was a specific use case that resulted in this change.

Reply:
I've just rewritten this section with some more specifics, although still no concrete examples. This change came out of a conversation with @stanford-scs, where we were attempting to make CAP-40 usable in cross-chain applications. As noted this is not practical when the cross-chain interaction requires signing a transaction for another chain because most chains do not hash the ed25519 payload before signing, which would require a large maximum length payload that is unreasonable (e.g. 256 to 512 bytes, or greater). However, @stanford-scs noted that if we simply increase the maximum payload to 64 bytes, we allow the payload to contain any hash since most hash algorithms that are defined today that produce digests that are over 32 bytes long, do not exceed 64 bytes. While this wouldn't support signing a transaction for another chain, it could allow signing a hash of some data that is verified by a smart contract on another chain and the change to 64 bytes makes CAP-40 with a wider variety of hashing algorithms.